### PR TITLE
Move imagedownload to script

### PR DESCRIPTION
**Pull request by firecyberice**
@StefanScherer @Govinda-Fichtner  do you have improvements?

connects to #98

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ services:
 language: bash
 script:
   - make shellcheck
+  - make SWARM_VERSION=1.2.2 CONSUL_VERSION=0.6.4 get-cluster-lab-images
```
Comment (StefanScherer) [inline .travis.yml]: I would like to remove this extra step. This should all be done with a simple "make sd-card". This should be possible to put the target as dependency to the Makefile:

```
sd-image: get-cluster-lab-images build
```
 
We also should put the versions out of the .travis.yml into another place.

File: Makefile [hunk 2]
```diff
@@ -3,6 +3,18 @@ default: build
 build:
 	docker build -t image-builder-rpi .
 
+get-cluster-lab-images:
+	mkdir -p builder/files/var/local/
```
Comment (StefanScherer) [inline Makefile]: What about adding a shell script doing all these steps? Then we have a place to put the version numbers into. So someone can build it without Travis locally.

**Comment by StefanScherer** [comment]
With these two changes it would LGTM! Travis is already green.

**Comment by firecyberice** [comment]
I would prefer to create a config file for all checksums and versions. So there is one file to change if sone versions need an upgrade.

**Comment by StefanScherer** [comment]
Yeah, just put it into a seperate file `config` with bash syntax and then `source config` it in the two or three scripts.

File: builder/test/image_spec.rb [hunk 3]
```diff
@@ -47,11 +47,11 @@
     let(:stdout) { run_mounted("ls /var/local").stdout }
 
     it "Consul export file exists" do
-      expect(stdout).to contain('rpi-consul_v0.6.4.tar.gz')
+      expect(stdout).to contain('rpi-consul_0.6.4.tar.gz')
     end
 
     it "Docker Swarm export file exists" do
-      expect(stdout).to contain('rpi-swarm_v1.2.2.tar.gz')
+      expect(stdout).to contain('rpi-swarm_1.2.2.tar.gz')
```
Comment (Govinda-Fichtner) [inline builder/test/image_spec.rb]: @firecyberice If we rename those files here we need to adjust the configuration in https://github.com/hypriot/image-builder-rpi/blob/master/builder/files/boot/device-init.yaml#L5-L6. Is there a reason why the renaming was necessary?

**Comment by firecyberice** [comment]
We need to clarify how our tags on github and on the docker hub look like. e.g. with leading v or without
in some cases ist is maybe not possible to add or remove the v for the ci

**Comment by StefanScherer** [comment]
As @firecyberice as mentioned tagging with drone's docker deploy plugin you cannot rename the tag as there is no script in between.

1. Check upstream Docker Hub tags: 
  * consul uses `v0.6.4` as Docker image tag with `v` -> https://hub.docker.com/r/library/consul/tags/
  * swarm uses `1.2.3` as tag without `v` -> https://hub.docker.com/r/library/swarm/tags/

So there seems to be two standards :-)

2. Check upstream GitHub tags:
  * consul uses `v0.6.4` as GitHub tag -> https://github.com/hashicorp/consul/releases
  * swarm uses `v1.2.3` as tag with `v`-> https://github.com/docker/swarm/releases

So swarm mixes both variants and consul uses `v` for both.

* We use the same schema for hypriot/rpi-swarm on Docker Hub as upstream swarm image -> without `v`.
* We use a different schema for hypriot/rpi-consul on Docker Hub as upstream consul image -> without `v`.

Regardless of this investigations:

@firecyberice Please fix all remaining `v1.2.2` filenames and `v0.6.4` filenames in this PR to make this PR work and fix tests. (eg . with CTRL+SHIFT+F in Atom finding/replacing all occurrences)

File: .travis.yml [hunk 4]
```diff
@@ -4,6 +4,7 @@ services:
 language: bash
 script:
   - make shellcheck
+  - make get-cluster-lab-images
```
Comment (StefanScherer) [inline .travis.yml]: I still wonder why this extra script call is needed. Should be done within the Makefile dependencies.
I don't know the exact order.
The community should be able to run `make sd-image` and the missing depencies should be made first. In this case downloading the cluster lab images.

I'll try to add the makefile dependency instead and trigger a Travis build with it.

**Comment by firecyberice** [comment]
I think this PR is ready to merge.
ping @StefanScherer  @Govinda-Fichtner

**Pull request merged**